### ISSUE TITLE
Mocking nested generics

### ISSUE THREAD

**Issue by Mayjest** [ffMathy/FluffySpoon.JavaScript.Testing.Faking#55]
AKA mocking a mongoose model. Not sure if nested generics is the right term.

For context: I'm playing around with a Nestjs app using mongoose & came across Substitute due to it's ability to provide mocks of interfaces, brilliant when dealing with mongoose documents. However, I'm having trouble mocking away the model and I believe it's due to nested generics. 

Here's a couple of bits of stripped down code to explain:

```
import { Document } from "mongoose";

export interface Cat extends Document {
  fieldA: string;
  fieldB: string;
}
````

```
import Substitute, { Arg } from "@fluffy-spoon/substitute";
import { Model } from "mongoose";
import { Cat } from "./cat";

const mockCat = Substitute.for<Cat>();
const mockCatModel = Substitute.for<Model<Cat>>();

mockCat.save().returns(Promise.resolve(mockCat));
mockCatModel.create().returns(); // valid
mockCatModel.create(Arg.all()).returns(); // Property 'returns' does not exist on type 'Promise<Cat>'
mockCatModel.create({fieldA: 'smith'}).returns(); // Property 'returns' does not exist on type 'Promise<Cat>
```

As you can see, I can't get it to mock out the create() method when a parameter is provided. Given it works in plenty of other places when it's just a `<Cat>`, I assume it's something to do with it being a `<Model<Cat>>`.

Also, I'm fairly new to typescript, so apologies if simple things like '`<Model<Cat>>` obviously wouldn't work because that's not how typescript works' have been missed :)

**Comment by Kosta-Github** [comment]
Maybe this is more related to optional arguments? see: https://github.com/ffMathy/FluffySpoon.JavaScript.Testing.Faking/issues/59

**Comment by ffMathy** [comment]
I suppose this was fixed in #59? If not, let me know.
